apps/controller/main.cpp

```diff
@@ -0,0 +1,132 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include "AnalogInputGroup.h"
+#include "DigitalInputGroup.h"
+#include "FrameController.h"
+#include "mbed.h"
+
+using namespace FrameProtocol;
+
+#define DEBUG
+
+static constexpr uint16_t SEND_INTERVAL_MS         = 100;
+static constexpr uint8_t DEBOUNCE_THRES            = 3;
+static constexpr uint8_t ANALOG_NUM_SAMPLE_AVERAGE = 5;
+
+static constexpr uint8_t NUM_BTNS     = 3;
+static constexpr uint8_t NUM_SWITCHES = 4;
+static constexpr uint8_t NUM_JOYS     = 4;
+static constexpr uint8_t NUM_POTS     = 2;
+
+BusIn digital_inputs_bus_btns(JOY_BTN, PB1, PB2);
+BusIn digital_inputs_bus_switches(SW_1A, SW_1B, SW_2A, SW_2B);
+
+// we can tune these parameters while testing
+DigitalInputGroup btns(digital_inputs_bus_btns, NUM_BTNS, InputDebounceType::INTERGRATING, DEBOUNCE_THRES);
+DigitalInputGroup switches(digital_inputs_bus_switches, NUM_SWITCHES, InputDebounceType::INTERGRATING, DEBOUNCE_THRES);
+
+FrameController::DigitalFrameConfig btns_frame_config = {
+    .boardType  = BoardType::MASTER,
+    .inputType  = DigitalInputType::BTN,
+    .inputGroup = btns,
+};
+
+FrameController::DigitalFrameConfig switches_frame_config = {
+    .boardType  = BoardType::MASTER,
+    .inputType  = DigitalInputType::SWITCH,
+    .inputGroup = switches,
+};
+
+AnalogBusIn analog_inputs_bus_joys(JOY_SM_X, JOY_SM_Y, JOY_X, JOY_Y);
+AnalogBusIn analog_inputs_bus_pots(POT_AL, SLIDE_POT_AL);
+
+AnalogInputGroup joys(analog_inputs_bus_joys, NUM_JOYS, ANALOG_NUM_SAMPLE_AVERAGE);
+AnalogInputGroup pots(analog_inputs_bus_pots, NUM_POTS, ANALOG_NUM_SAMPLE_AVERAGE);
+
+FrameController::AnalogFrameConfig joys_frame_config = {
+    .boardType  = BoardType::MASTER,
+    .inputType  = AnalogInputType::JOY,
+    .inputGroup = joys,
+};
+
+FrameController::AnalogFrameConfig pots_frame_config = {
+    .boardType  = BoardType::MASTER,
+    .inputType  = AnalogInputType::POT,
+    .inputGroup = pots,
+};
+
+Timer timer;
+
+// Serial object for sending data to PC
+RawSerial pc(USBTX, USBRX);
```

Comment:
Is this board using the USB interface directly?

Reply:
What do you mean by using the USB interface directly? 
What I know is that we will not be using the USB protocol, we are using this IC (https://www.ftdichip.com/Support/Documents/DataSheets/ICs/DS_FT230X.pdf) which converts USB to UART...so under the hood we are just using UART

Reply:
Ok that makes sense